base/src/main/kotlin/net/corda/v5/base/types/MemberX500Name.kt

```diff
@@ -9,39 +9,130 @@ import javax.naming.ldap.Rdn
 import javax.security.auth.x500.X500Principal
 
 /**
- * X.500 distinguished name data type customised to how Corda uses names. This restricts the attributes to those Corda
+ * X.500 distinguished name data type customised to how Corda membership uses names. This restricts the attributes to those Corda
  * supports, and requires that organisation, locality and country attributes are specified. See also RFC 4519 for
  * the underlying attribute type definitions.
  *
  * The class also guaranties the reliable equality comparison regardless which order the attributes are specified when
  * parsing from the string or X500principal as well outputs the attributes to string in predictable order.
  *
- * @property commonName optional name by the which the entity is usually known. Used only for services (for
- * organisations, the [organisation] property is the name). Corresponds to the "CN" attribute type.
- * @property organisationUnit optional name of a unit within the [organisation]. Corresponds to the "OU" attribute type.
- * @property organisation name of the organisation. Corresponds to the "O" attribute type.
- * @property locality locality of the organisation, typically the nearest major city. For distributed services this would be
- * where one of the organisations is based. Corresponds to the "L" attribute type.
- * @property state the full name of the state or province the organisation is based in. Corresponds to the "ST"
+ * There may be additional network specific requirements which need to be taken into account when creating a name.
```

Comment:
I don’t particular like that statement

```
 * There may be additional network specific requirements which need to be taken into account when creating a name.
 * For example, the network operator may require a particular format for names so that they can issue suitable certificates.
 ```

As it implies that the class can do something about that but in reality that is something which is external to this class, so if you reword it in a way that is clear that it must be done outside that class.